src/object/README.md

```diff
@@ -1,261 +1,145 @@
+  For example, if an application linked to the DAOS client library crashed, then
+  it should either replay uncompleted I/O requests if it has log for those I/Os,
+  or abort the uncommitted epoch.
 
-Furthermore, when failure occurs, the client replication protocol still needs some extra mechanism to enforce consistency between replicas:
-
-- If the DAOS-SR client can capture the failure, e.g. a target failed during I/O, because the DAOS-SR client can complete this I/O by switching to another target and resubmitting request. At the meanwhile, the DAOS-SR servers can rebuild the missing replica in the background. Therefore, DAOS-SR can still guarantee data consistency between replicas.
-- If DAOS-SR cannot capture the failure, for example, the DAOS-SR client itself crashed before successfully updating all replicas so some replicas may lag behind. Based on the current replication protocol, the DAOS-SR servers cannot detect the missing I/O requests, so DAOS-SR cannot guarantee data consistency between replicas. The upper layer stack has to either re-submit the interrupted I/O requests to enforce data consistency between replicas, or abort the epoch and rollback to the consistent status of the container. For example, if an application linked to the SR client library crashed, then it should either replay uncompleted I/O requests if it has log for those I/Os, or abort the uncommitted epoch.
-
-<a id="10.4.1.2"></a>
 #### Server replication
 
-DAOS-SR could also support server replication, which has stronger consistency of replicas with a trade-off in performance and latency, for example, a primary-copy replication protocol. With a server-side protocol, because the client only sends updates to the primary replica, conflict writes should be detected and serialized by the server, and it is safe to have overlapped writes in the same epoch.
+DAOS also supports server replication, which has stronger consistency of
+replicas with a trade-off in performance and latency. In server replication mode
+DAOS client selects a leader shard to send the IO request with the need-to-
+forward shards embedded in the RPC request, when the leader shard gets that IO
+request it handles it as below steps:
+- firstly forward the IO request to others shards
+  For the request forwarding, it is offload to the vos target's offload extream
+  to release the main IO service extream from IO request sending and reply
+  receiving (see shard_req_forward).
+- then serve the IO request locally
+- wait the forwarded IO's completion and reply client IO request.
+
+In this mode the conflict writes can be detected and serialized by the leader
+shard server. Now both modes are supported by DAOS, it can be dynamically
+configured by environment variable "DAOS_IO_SRV_DISPATCH" before loading DAOS
+server. By default DAOS works in server replication mode, and if the ENV set as
+zero then will work in client replication mode.
```

Comment:
I did not see retry I/O is mentioned somewhere? that is a important part of src/object/